test/test-collection.js

```diff
@@ -24,25 +24,56 @@ function mockTest(opts, title) {
 }
 
 function titles(tests) {
+	if (!tests) {
+		tests = [];
+	}
+
 	return tests.map(function (test) {
 		return test.title;
 	});
 }
 
+function removeEmptyProps(obj) {
+	if (Array.isArray(obj) && obj.length === 0) {
+		return null;
+	}
+
+	if (obj.constructor !== Object) {
```

Comment:
Why do we care what the constructor is? Are we just trying to avoid cleaning primitives?